client/tests/BasicModal.test.jsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import BasicModal from '../components/common/BasicModal';
+
+const MockCustomComponent = ({ customProp }) => <button>{customProp}</button>; // eslint-disable-line react/prop-types
+
+describe('BasicModal', () => {
+    test('renders modal when show is true', () => {
+        render(
+            <BasicModal show={true} title="Test title" content="Test content" />
+        );
+        expect(screen.getByRole('dialog')).toBeInTheDocument();
+    });
+
+    test('does not render modal when show is false', () => {
+        render(
+            <BasicModal
+                show={false}
+                title="Test title"
+                content="Test content"
+            />
+        );
+        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    });
+
+    test('renders title and content correctly', () => {
+        render(
+            <BasicModal show={true} title="My Title" content="My Content" />
+        );
+        expect(screen.getByText('My Title')).toBeInTheDocument();
+        expect(screen.getByText('My Content')).toBeInTheDocument();
+    });
+
+    test('multiple action buttons trigger correct functions', () => {
+        const saveFunction = jest.fn();
+        const deleteFunction = jest.fn();
+        render(
+            <BasicModal
+                show={true}
+                title="Test title"
+                content="Test content"
+                actions={[
+                    { label: 'Save', onClick: saveFunction },
+                    { label: 'Delete', onClick: deleteFunction }
+                ]}
+            />
+        );
+        fireEvent.click(screen.getByText('Save'));
+        fireEvent.click(screen.getByText('Delete'));
+        expect(saveFunction).toHaveBeenCalledTimes(1);
```

Comment:
Let's swap these two lines (or better yet, duplicate the assertion) so the test verifies the correct label-onClick association.

Reply:
Wow! Color me impressed that you detected that.